perfkitbenchmarker/sample.py

```diff
@@ -57,12 +58,57 @@ class Sample(collections.namedtuple('Sample', _SAMPLE_FIELDS)):
     value: float. Result for 'metric'.
     unit: string. Units for 'value'.
     metadata: dict. Additional metadata to include with the sample.
+    timestamp: float. Unix timestamp.
   """
 
-  def __new__(cls, metric, value, unit, metadata=None, **kwargs):
+  def __new__(cls, metric, value, unit, metadata=None, timestamp=None,
+              **kwargs):
+    if timestamp is None:
+      timestamp = time.time()
+
     return super(Sample, cls).__new__(cls, metric, value, unit,
-                                      metadata=metadata or {}, **kwargs)
+                                      metadata=metadata or {},
+                                      timestamp=timestamp,
+                                      **kwargs)
 
   def asdict(self):
     """Converts the Sample to a dictionary."""
     return self._asdict()
+
+
+class SamplesTestMixin(object):
```

Comment:
I'm working on this, but I don't think tests/ is in the load path for our test modules when they run. Is there some place in perfkitbenchmarker/ that would work for this? Maybe a new folder called test_util/ ?